### PR TITLE
memcached-operator: add memcached-operator example

### PR DESCRIPTION
**Pull request by AlexNPavel**
This is based on the example in the operator-sdk user-guide and also
includes example tests using the test framework.

### CODE REVIEW

**Comment by fanminshi** [comment]
@AlexNPavel could you add a readme on demonstrating the newly added testing framework and the testing command.

File: memcached-operator/test/e2e/memcached_test.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+package e2e
+
+import (
+	goctx "context"
+	"fmt"
+	"testing"
+	"time"
+
+	operator "github.com/operator-framework/operator-sdk/test/test-framework/pkg/apis/cache/v1alpha1"
```
Comment (hasbro17) [inline memcached-operator/test/e2e/memcached_test.go]: This should be using the memcached CR type from within it's own project:
```
cachev1alpha1 "github.com/operator-framework/operator-sdk-samples/pkg/apis/cache/v1alpha1"
```

File: memcached-operator/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- [dep][dep_tool] version v0.5.0+.
+- [go][go_tool] version v1.10+
+- [docker][docker_tool] version 17.03+
+- Access to a kubernetes v1.9.0+ cluster
+
+### Install the Operator SDK CLI
+First, checkout and install the operator-sdk CLI:
+```
+$ cd $GOPATH/src/github.com/operator-framework/operator-sdk
+$ git checkout master // currently, there are not releases that include the test framework, so use the master for now
```
Comment (hasbro17) [inline memcached-operator/README.md]: there are no releases

File: memcached-operator/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+$ cd $GOPATH/src/github.com/operator-framework/operator-sdk
+$ git checkout master // currently, there are no releases that include the test framework, so use the master for now
+$ dep ensure
+$ go install github.com/operator-framework/operator-sdk/commands/operator-sdk
+```
+
+### Initial Setup
+Checkout this Memcached Operator repository
+```
+$ mkdir $GOPATH/src/github.com/operator-framework
```
Comment (hasbro17) [inline memcached-operator/README.md]: This should already be present if they installed the sdk binary from `github.com/operator-framework/operator-sdk` in the previous step. So we can remove this.

**Comment by hasbro17** [comment]
LGTM

**Pull request merged**